File: cpp-tiny/lib/Models/BadEventQueueIdErrorAllOf.cpp


#include "BadEventQueueIdError_allOf.h"

using namespace Tiny;

BadEventQueueIdError_allOf::BadEventQueueIdError_allOf()
{
	result = null;
	msg = null;
	code = null;
	queue_id = std::string();
}

BadEventQueueIdError_allOf::BadEventQueueIdError_allOf(std::string jsonString)
{
	this->fromJson(jsonString);
}

BadEventQueueIdError_allOf::~BadEventQueueIdError_allOf()
{

}

void
BadEventQueueIdError_allOf::fromJson(std::string jsonObj)
{   
    bourne::json object = bourne::json::parse(jsonObj);

    const char *resultKey = "result";
    
    if(object.has_key(resultKey)) 
    {
        bourne::json value = object[resultKey];


        

        AnyType* obj = &result;
		obj->fromJson(value.dump());

    }

    const char *msgKey = "msg";
    
    if(object.has_key(msgKey)) 
    {
        bourne::json value = object[msgKey];


        

        AnyType* obj = &msg;
		obj->fromJson(value.dump());

    }

    const char *codeKey = "code";
    
    if(object.has_key(codeKey)) 
    {
        bourne::json value = object[codeKey];


        

        AnyType* obj = &code;
		obj->fromJson(value.dump());

    }

    const char *queue_idKey = "queue_id";
    
    if(object.has_key(queue_idKey)) 
    {
        bourne::json value = object[queue_idKey];


        
        jsonToValue(&queue_id, value, "std::string");


    }


}

bourne::json
BadEventQueueIdError_allOf::toJson()
{
    bourne::json object = bourne::json::object();

    




	object["result"] = getResult().toJson();

    




	object["msg"] = getMsg().toJson();

    




	object["code"] = getCode().toJson();

    



    object["queue_id"] = getQueueId();



    return object;

}

AnyType
BadEventQueueIdError_allOf::getResult()
{
	return result;
}

void
BadEventQueueIdError_allOf::setResult(AnyType  result)
{
	this->result = result;
}

AnyType
BadEventQueueIdError_allOf::getMsg()
{
	return msg;
}

void
BadEventQueueIdError_allOf::setMsg(AnyType  msg)
{
	this->msg = msg;
}

AnyType
BadEventQueueIdError_allOf::getCode()
{
	return code;
}

void
BadEventQueueIdError_allOf::setCode(AnyType  code)
{
	this->code = code;
}

std::string
BadEventQueueIdError_allOf::getQueueId()
{
	return queue_id;
}

void
BadEventQueueIdError_allOf::setQueueId(std::string  queue_id)
{
	this->queue_id = queue_id;
}

File: spring/src/main/java/org/openapitools/model/BadEventQueueIdErrorAllOf.java
package org.openapitools.model;

import java.util.Objects;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonCreator;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import org.openapitools.jackson.nullable.JsonNullable;
import javax.validation.Valid;
import javax.validation.constraints.*;

/**
 * BadEventQueueIdErrorAllOf
 */
@javax.annotation.Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2021-08-08T20:43:06.633496Z[Etc/UTC]")
public class BadEventQueueIdErrorAllOf   {
  @JsonProperty("result")
  private JsonNullable<Object> result = JsonNullable.undefined();

  @JsonProperty("msg")
  private JsonNullable<Object> msg = JsonNullable.undefined();

  @JsonProperty("code")
  private JsonNullable<Object> code = JsonNullable.undefined();

  @JsonProperty("queue_id")
  private String queueId;

  public BadEventQueueIdErrorAllOf result(Object result) {
    this.result = JsonNullable.of(result);
    return this;
  }

  /**
   * Get result
   * @return result
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getResult() {
    return result;
  }

  public void setResult(JsonNullable<Object> result) {
    this.result = result;
  }

  public BadEventQueueIdErrorAllOf msg(Object msg) {
    this.msg = JsonNullable.of(msg);
    return this;
  }

  /**
   * Get msg
   * @return msg
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getMsg() {
    return msg;
  }

  public void setMsg(JsonNullable<Object> msg) {
    this.msg = msg;
  }

  public BadEventQueueIdErrorAllOf code(Object code) {
    this.code = JsonNullable.of(code);
    return this;
  }

  /**
   * Get code
   * @return code
  */
  @ApiModelProperty(value = "")


  public JsonNullable<Object> getCode() {
    return code;
  }

  public void setCode(JsonNullable<Object> code) {
    this.code = code;
  }

  public BadEventQueueIdErrorAllOf queueId(String queueId) {
    this.queueId = queueId;
    return this;
  }

  /**
   * The string that identifies the invalid event queue. 
   * @return queueId
  */
  @ApiModelProperty(value = "The string that identifies the invalid event queue. ")


  public String getQueueId() {
    return queueId;
  }

  public void setQueueId(String queueId) {
    this.queueId = queueId;
  }


  @Override
  public boolean equals(Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    BadEventQueueIdErrorAllOf badEventQueueIdErrorAllOf = (BadEventQueueIdErrorAllOf) o;
    return Objects.equals(this.result, badEventQueueIdErrorAllOf.result) &&
        Objects.equals(this.msg, badEventQueueIdErrorAllOf.msg) &&
        Objects.equals(this.code, badEventQueueIdErrorAllOf.code) &&
        Objects.equals(this.queueId, badEventQueueIdErrorAllOf.queueId);
  }

  @Override
  public int hashCode() {
    return Objects.hash(result, msg, code, queueId);
  }

  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class BadEventQueueIdErrorAllOf {\n");
    
    sb.append("    result: ").append(toIndentedString(result)).append("\n");
    sb.append("    msg: ").append(toIndentedString(msg)).append("\n");
    sb.append("    code: ").append(toIndentedString(code)).append("\n");
    sb.append("    queueId: ").append(toIndentedString(queueId)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }
}

